test/parallel/test-stream-pipe-await-drain-push-while-write.js

```diff
@@ -0,0 +1,25 @@
+'use strict';
+const stream = require('stream');
+const common = require('../common');
+
+// A writable stream which pushes data onto the stream which pipes into it,
+// but only the first time it's written to. Since it's not paused at this time,
+// a second write will occur. If the pipe increases awaitDrain twice, we'll
+// never get subsequent chunks because 'drain' is only emitted once.
+const writable = new stream.Writable();
+writable._write = common.mustCall((chunk, encoding, cb) => {
+  if (chunk.length === 32 * 1024) { // first chunk
+    pthru.push(new Buffer(33 * 1024)); // above hwm
+    return process.nextTick(cb); // let pipe increment awaitDrain
+  }
+  cb();
+}, 3);
+
+const pthru = new stream.PassThrough();
```

Comment:
You could use `new stream.Readable({read: () => {}})`, but you can just leave it as a `PassThrough` if you prefer, I really was just curious. :)